source/Calamari.AzureWebApp/Integration/Websites/Publishing/ResourceManagerPublishProfileProvider.cs

```diff
@@ -36,18 +36,8 @@ public async Task<WebDeployPublishSettings> GetPublishProperties(IAzureAccount a
                 log.InfoFormat("Using override for Azure Active Directory endpoint - {0}", account.ActiveDirectoryEndpointBaseUri);
 
             var token = account.AccountType == AccountType.AzureOidc
-                ? await AzureOidcAccountExtensions.GetAuthorizationToken(account.TenantId,
-                                                                               account.ClientId,
-                                                                               account.GetCredentials,
-                                                                               account.ResourceManagementEndpointBaseUri,
-                                                                               account.ActiveDirectoryEndpointBaseUri,
-                                                                               account.AzureEnvironment,
-                                                                               CancellationToken.None)
-                : await AzureServicePrincipalAccountExtensions.GetAuthorizationToken(account.TenantId,
-                                                                                     account.ClientId,
-                                                                                     account.GetCredentials,
-                                                                                     account.ResourceManagementEndpointBaseUri,
-                                                                                     account.ActiveDirectoryEndpointBaseUri);
+                ? await (account as AzureOidcAccount).GetAuthorizationToken(CancellationToken.None)
+                : await (account as AzureServicePrincipalAccount).GetAuthorizationToken();
```

Comment:
Simplifies the method calls. But also makes it easier for the discovery endpoint to travel downstream.